server/tasks/jobs/sendReports.js

```diff
@@ -0,0 +1,49 @@
+const { EmailReportsFrequency } = require('../../common/variables');
+const { getItemsForReport } = require('../../common/utils');
+const { sendReport } = require('../../mailer');
+const User = require('../../models/user.model');
+const List = require('../../models/list.model');
+
+const days = {
+  0: EmailReportsFrequency.SUNDAY,
+  1: EmailReportsFrequency.MONDAY,
+  2: EmailReportsFrequency.TUESDAY,
+  3: EmailReportsFrequency.WEDNESDAY,
+  4: EmailReportsFrequency.THURSDAY,
+  5: EmailReportsFrequency.FRIDAY,
+  6: EmailReportsFrequency.SATURDAY
+};
+
+const sendReports = (agenda, jobName) => {
+  agenda.define(jobName, async job => {
+    const day = new Date().getDay();
+    const reportDay = days[day];
+    const { HOST } = process.env;
+
+    try {
+      const users = await User.find(
+        { emailReportsFrequency: reportDay },
+        'displayName email'
+      )
+        .lean()
+        .exec();
+
+      users.forEach(async user => {
+        try {
+          const report = await getItemsForReport(List, user);
+
+          if (report) {
+            await sendReport(HOST, report);
+          }
+        } catch {
+          // Ignore errors
+        }
+      });
```

Comment:
Check if you can get rid of this nested try catch to simplify this a little bit by using our helper fireAndForget()


```
[].forEach(async user => {
  const report = await fireAndForget(getItemsForReport(List, user));

  if (report) {
    fireAndForget(sendReport(HOST, report));
  }
});

```

Reply:
This doesn't solve the problem because there is no problem with `sendReport` but with `getItemsForReport`. If there is no nested `try-catch` and the `getItemsForReport` function fails the whole forEach will fail. Of course the last `catch` catches the error but we won't send reports to the rest of users. 

We can't add `fireAndForget` to `getItemsForReport` because we wait for the result. 

But I added `fireAndForget(sendReport(HOST, report))` to have additional protection against sending errors.